.github/workflows/build.yml

```diff
@@ -31,6 +31,37 @@ jobs:
       - name: leanpkg build
         run: leanpkg build | python scripts/detect_errors.py
 
+      - name: configure git setup
+        run: |
+          git remote add origin-bot "https://leanprover-community-bot:${{ secrets.DEPLOY_NIGHTLY_GITHUB_TOKEN }}@github.com/leanprover-community/mathlib.git"
+          git config user.email "leanprover.community@gmail.com"
+          git config user.name "leanprover-community-bot"
+
+          # By default, github actions overrides the credentials used to access any
+          # github url so that it uses the github-actions[bot] user.  We want to access
+          # github using a different username.
+          git config --unset http.https://github.com/.extraheader
+
+      - name: push release to azure
+        if: always() && github.repository == 'leanprover-community/mathlib' && github.event_name == 'push' && matrix.lean_version == 'leanprover-community/lean:3.5.1'
```

Comment:
Should we factor out all the appearances of `3.5.1` to some global variable, for easy of updating?

Reply:
Now that we only build with the Lean version specified in `leanpkg.toml`, we should just remove the version everywhere (including the `elan override set`).

Reply:
Yes, this is a good idea. Should this PR become a general CI refactor, or should I do that separately?

Reply:
The string `3.5.1` appears in 8 mathlib files. 6 look relevant and the others can probably be removed now. I think we can "automate" the version upgrade by just replacing the string.

Reply:
> Should this PR become a general CI refactor, or should I do that separately?

Whichever you prefer.

Reply:
Let's do it separately. This is tested and working (AFAICT), I'll get the rest working independently.